Feature/oauth

sergeJAVA opened the pull request.

sergeJAVA committed:
feat: добавил OAuth2

Пока пусть будет так. Надо подправить, плюс тест сломался.

sergeJAVA committed:
feat: доделал OAuth2 и login.html форму

починил сломанные тесты

sergeJAVA committed:
chore: Убрал зависимость thymeleaf

Убрал, потому что реализовал форму через html + javascript, а не через шаблонизатор.

sergeJAVA committed:
refactor:

AlKl1M commented on src/main/java/com/webbee/auth_service_webbee/handler/OAuth2AuthenticationSuccessHandler.java:
```diff
@@ -0,0 +1,71 @@
+        }
+
+        String token = jwtService.generateJwtToken(user);
+        String targetUrl = "http://localhost:8081/oauth2/callback?token=" + token;
```
Хардкод, выносим в параметры

AlKl1M commented on src/main/resources/static/login.html:
```diff
@@ -0,0 +1,69 @@
+</div>
+
+<script>
+    document.getElementById("loginForm").addEventListener("submit", async function (e) {
```
Де факто докапываюсь, но можно внести в js файл.

AlKl1M reviewed (changes requested):
Нужны интеграционные тесты - у тебя есть банально контроллеры, которые хорошо было бы протестить интеграционными. Хотяб на 50%

sergeJAVA committed:
refactor: убрал хардкод url'а

sergeJAVA committed:
test: Добавил интеграционные тесты.

Добавил интеграционные тесты для контроллеров, а также исправил баг, вызванный ошибкой в логике метода getUserNameFromToken в JwtService.

sergeJAVA commented:
80% покрытие тестов получилось

AlKl1M reviewed (approved):
Норм, вливай.
Если у Артема будут замечания - фиксом подольешь.
